File: app/src/main/java/com/telugucalendar/telugupanchangamr/Adapters/PushkaraluAdapter.java
package com.telugucalendar.telugupanchangamr.Adapters;

import android.app.Activity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.telugucalendar.telugupanchangamr.Model.Gowri;
import com.telugucalendar.telugupanchangamr.Model.Pushkaralu;
import com.telugucalendar.telugupanchangamr.Model.PushkaraluVariant;
import com.telugupanchangam.telugucalender.R;

import java.util.ArrayList;


public class PushkaraluAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    final Activity activity;
    ArrayList<Pushkaralu> pushkaralus;

    public PushkaraluAdapter(Activity activity, ArrayList<Pushkaralu> pushkaralus) {
        this.activity = activity;
        this.pushkaralus = pushkaralus;
    }
    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(activity).inflate(R.layout.pushkaralu_tab_layout, parent, false);
        return new ExploreItemHolder(view);
    }



    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holderParent, int position) {
        final ExploreItemHolder holder = (ExploreItemHolder) holderParent;
        final Pushkaralu pushkaralu = pushkaralus.get(position);

        Log.d("PushkaraluAdapter", "Binding item at position: " + position + " with title: " + pushkaralu.getTitle());

        if (pushkaralu.getPushkaralu_variant() != null && !pushkaralu.getPushkaralu_variant().isEmpty()) {
            PushkaraluVariant variant = pushkaralu.getPushkaralu_variant().get(0);

            holder.wbsub_title.loadDataWithBaseURL("file:///android_asset/fonts/", "<html>\n" +
                    "<head>\n" +
                    "<link rel='stylesheet' type='text/css' href='file:///android_asset/fonts/ramabhadra.css'>" +
                    "<style>\n" +
                    "body {\n" +
                    "    font-family: 'ramabhadra';\n" +
                    "    text-align: center;\n" +
                    "}\n" +
                    "</style>" +
                    "</head>" +
                    "<body>\n" +
                    variant.getSub_title() +
                    "</body>\n" +
                    "</html>", "text/html", "UTF-8", "");

            holder.wbsub_description.loadDataWithBaseURL("file:///android_asset/fonts/", "<html>\n" +
                    "<head>\n" +
                    "<link rel='stylesheet' type='text/css' href='file:///android_asset/fonts/ramabhadra.css'>" +
                    "<style>\n" +
                    "body {\n" +
                    "    font-family: 'ramabhadra';\n" +
                    "    text-align: center;\n" +
                    "}\n" +
                    "</style>" +
                    "</head>" +
                    "<body>\n" +
                    variant.getSub_description() +
                    "</body>\n" +
                    "</html>", "text/html", "UTF-8", "");
        }
    }


    @Override
    public int getItemCount()
    {
        return pushkaralus.size();
    }

    static class ExploreItemHolder extends RecyclerView.ViewHolder {
        final WebView wbsub_title,wbsub_description;

        public ExploreItemHolder(@NonNull View itemView) {
            super(itemView);
            wbsub_title = itemView.findViewById(R.id.wbsub_title);
            wbsub_description = itemView.findViewById(R.id.wbsub_description);
          


        }
    }
}

File: app/src/main/java/com/telugucalendar/telugupanchangamr/Adapters/PushkaraluAdapter.kt
package com.telugucalendar.telugupanchangamr.Adapters

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.telugucalendar.telugupanchangamr.Model.Festival
import com.telugucalendar.telugupanchangamr.Model.Grahalu
import com.telugucalendar.telugupanchangamr.Model.Pushkaralu
import com.telugupanchangam.telugucalender.R


import java.util.*



class PushkaraluAdapter(val activity: Activity, pushkaralu: ArrayList<Pushkaralu>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    var pushkaralu: ArrayList<Pushkaralu>

    init {
        this.pushkaralu = pushkaralu
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view: View =
            LayoutInflater.from(activity).inflate(R.layout.moudya_tab_layout, parent, false)
        return ExploreItemHolder(view)
    }

    override fun onBindViewHolder(holderParent: RecyclerView.ViewHolder, position: Int) {
        val holder = holderParent as ExploreItemHolder
        val pushkaralu: Pushkaralu = pushkaralu[position]

        holder.tvTitle.text = pushkaralu.sub_title
        holder.tvDescription.text = pushkaralu.sub_description



    }
    override fun getItemCount(): Int {
        return pushkaralu.size
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }

    internal class ExploreItemHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        val tvTitle: TextView
        val tvDescription: TextView


        init {
            tvTitle = itemView.findViewById(R.id.tvTitle)
            tvDescription = itemView.findViewById(R.id.tvDescription)
        }
    }
}